src/main/scala/GithubRelease.scala

```diff
@@ -57,48 +50,34 @@ case object GithubRelease {
       typeMap.getContentType
     }
 
-    private def readCredentialsFrom(file: File): Option[GitHubCredentials] = {
+    private def readCredentialsFrom(file: File): Option[String] = {
       val credentialsFile = Option(file)
         .filter(_.isFile)
         .filter(_.canRead)
 
       val maybeCredentialParameters = credentialsFile.map { githubCredentialsFile =>
-        val lines = IO.readLines(githubCredentialsFile)
-        lines map { line =>
-          line.split("=")
-        } filter {
-          _.length == 2
-        } map { lineElements =>
-          lineElements(0) -> lineElements(1)
-        } toMap
+        import scala.collection.JavaConverters._
+        val props = new java.util.Properties()
+        props.load(new java.io.FileInputStream(githubCredentialsFile))
+        props.asScala 
       }
 
-      maybeCredentialParameters.flatMap { credentialParameters =>
-        val login = for {
-          login <- credentialParameters.get("login")
-          password <- credentialParameters.get("password")
-        } yield GitHubLogin(login, password)
-
-        val token = credentialParameters.get("oauth").map(GitHubToken)
-        token orElse login
-      }
+      maybeCredentialParameters.flatMap( _.get("oauth"))
     }
 
-    def ghreleaseGetCredentials: DefTask[GitHubCredentials] = Def.task {
+    def ghreleaseGetCredentials: DefTask[String] = Def.task {
       val log = streams.value.log
 
       val conf = file(System.getProperty("user.home")) / ".github"
-      ghreleaseGithubCreds.value orElse readCredentialsFrom(conf) getOrElse {
+
+      ghreleaseGithubToken.value orElse readCredentialsFrom(conf) getOrElse {
         sys.error("If you want to use sbt-github-release plugin, you should set credentials correctly")
       }
     }
```

Comment:
I think that there's no point in having both things: a setting for the token and a task. The token has to be loaded once (per build), so it perfectly can be a setting. You can make `readCredentialsFromFile` public, add a similar method for env-var and set the default value of `ghreleaseGithubToken` to "try to read from env-var, then try to read from the default file or fail with an informative error message". User can change it with his own retrieval logic and set the value explicitly or read from another file/env-var.